### ISSUE TITLE
Default values for string Input prompt are not passed to validators

### ISSUE THREAD

**Issue by agc93** [shibayan/Sharprompt#84]
If a `defaultValue` is specified in a `Prompt.Input<string>` call that also specifies validators, pressing Enter on the prompt sends an empty string to the validators, not the default value. For example:

```csharp
var userName = Sharprompt.Prompt.Input<string>("Please enter a name", "default", validators: new[] { Sharprompt.Validations.Validators.Required()});
```

will fail unless providing a non-default value, since an empty string is passed to validators, rather than the default value specified by `defaultValue`.

---

I've created an oversimplified repro [in this repo](https://github.com/whyareyoudoingthis/sharprompt-defaults)

**Comment by shibayan** [comment]
Thanks for the report. This is a bug and I will fix it.

**Comment by shibayan** [comment]
Fixed `v2.1.0-preview2`

**Comment by jods4** [comment]
@shibayan this doesn't impact only the string Input.
For example the Confirm input clearly shows a default N value in UI:
```
Are you sure? (y/N)
```
But if you proceed with hitting ENTER the validator says a value is required.

**Comment by shibayan** [comment]
@jods4 
Thanks for the feedback. In Confirm, the "(y/N)" display is now fixed value, so that's how it appears.
https://github.com/shibayan/Sharprompt/blob/master/Sharprompt/Forms/Confirm.cs#L66

Are the following mappings commonly used? I'm not familiar with it so please let me know if you know.

- (y/N) = default value is `false`
- (Y/n) = default value is `true`
- (y/n) = not default value (required input)

**Comment by jods4** [comment]
@shibayan For me (y/N) is a good english default for a confirm.
My gripe is that when I hit ENTER it doesn't work because the default is not passed to the validator, as for the text. input

**Comment by shibayan** [comment]
There are no validators in Confirm. It is required, but can be omitted if a default value is explicitly specified (i.e., it passes by typing the Enter key).

So the behavior is correct, it's just that the message displayed is not as common as it should be.

**Comment by jods4** [comment]
It doesn't work for us, so I guess it's because the "implicit" default value is not taken into account -- or maybe the default message is just wrong given there's no default?

Displaying `(y/N)` is good because it shows that `y` or `n` is expected. Uppercase `N` means it's the default value if you just hit enter.

Are you saying that Confirm shows `(y/N)` although there is no default value?
So if I just explicitely add `default: false` it will work as intended?

**Comment by shibayan** [comment]
>Are you saying that Confirm shows (y/N) although there is no default value?

Yes

>So if I just explicitely add `default: false` it will work as intended?

Yes

**Comment by shibayan** [comment]
The message display will be corrected in #87.
